### PR TITLE
refactor(ios/engine): chained lexical model install callback, fix

### PR DESCRIPTION
**Pull request by jahorton**
Looks like I made an invalid assumption in #3383 - that download-link format does not appear to support lexical model packages.

Past that, while working on the keyboard search, I noticed that there was a major issue with attempting to automate tests involving 'chained' lexical model downloads - cases where we've been automatically installing lexical models.  (In particular, we don't receive a callback after the lexical model installation occurs, making the standard `XCTestExpectation` synchronization scheme impossible to use properly.)  This PR aims to remedy that, which will also help `KeymanEngine` users be able to control if, when, and how this behavior occurs.

The existing behavior will be kept intact for legacy methods within this PR, while #3384 (with its newer methods) will aim to use a closure-based scheme to ensure that framework users may gain control of the process if desired.

### CODE REVIEW

File: ios/engine/KMEI/KeymanEngine/Classes/KeymanHosts.swift [hunk 1]
```diff
@@ -64,9 +64,16 @@ public enum KeymanHosts {
...
-      // TODO:  Once the staging-site changes are in place,
-      //        add switch-case here to select appropriate URL.
-      return URL.init(string: "https://keyman.com")!
+      // Necessary for development of the keyboard search, even though the staging site
+      // URL isn't quite stable yet.
+      switch forTier {
+        case .alpha:
+          fallthrough
+        case .beta:
+          return URL.init(string: "https://staging-keyman-com.azurewebsites.net")!
```
Comment (mcdurdin) [inline ios/engine/KMEI/KeymanEngine/Classes/KeymanHosts.swift:73]: I think keyman-staging.com is now stable 😄

Comment (jahorton) [reply ios/engine/KMEI/KeymanEngine/Classes/KeymanHosts.swift:73]: In that case, I might just slip in the necessary `switch-case`s for all three... assuming that holds for all 3?  Or just keyman-staging.com?

Comment (mcdurdin) [reply ios/engine/KMEI/KeymanEngine/Classes/KeymanHosts.swift:73]: `keyman-staging.com` is replacing `staging-keyman-com.azurewebsites.net`.

Comment (jahorton) [reply ios/engine/KMEI/KeymanEngine/Classes/KeymanHosts.swift:73]: Right.  What about `api.keyman-staging.com` and `help.keyman-staging.com`?  Are those ready now, or do they come later?

Comment (mcdurdin) [reply ios/engine/KMEI/KeymanEngine/Classes/KeymanHosts.swift:73]: > What about `api.keyman-staging.com` and `help.keyman-staging.com`? Are those ready now, or do they come later?

Yes, they are ready now as well.

**Pull request merged**